src/sentry/models/externalactor.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from django.db import models
+
+from sentry.db.models import BoundedPositiveIntegerField, DefaultFieldsModel, FlexibleForeignKey
+from sentry.types.integrations import ExternalProviders
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalActor(DefaultFieldsModel):
+    __core__ = False
+
+    actor = FlexibleForeignKey("sentry.Actor", db_index=True, on_delete=models.CASCADE)
+    organization = FlexibleForeignKey("sentry.Organization")
+    integration = FlexibleForeignKey("sentry.Integration", null=True)
+    provider = BoundedPositiveIntegerField(
+        choices=(
+            (ExternalProviders.EMAIL, "email"),
+            (ExternalProviders.SLACK, "slack"),
+            (ExternalProviders.MSTEAMS, "msteams"),
+            (ExternalProviders.PAGERDUTY, "pagerduty"),
+            (ExternalProviders.GITHUB, "github"),
+            (ExternalProviders.GITLAB, "gitlab"),
+        ),
+    )
+
+    # external name => display name i.e. username, team name, channel name
+    # external id => unique identifier i.e user id, channel id
+    external_name = models.TextField()
+    external_id = models.TextField(null=True)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_externalactor"
+        unique_together = (("organization", "provider", "external_name"),)
```

Comment:
I think this will prevent a user from linking Slack usernames in multiple workspaces in an organizations. I think there's also a problem with using a nullable field in a `unique_together`.

Reply:
@mgaeta @ceorourke what if it was unique together with the actor_id as well? I could see the external_name being the same for different users, but I think it makes sense to limit a user to just having one workspace per org (like for workflow notifications for the user)